code/src/mp3_player.cpp

```diff
@@ -0,0 +1,98 @@
+#include "mp3_player.hpp"
+
+#include <cstring>
+#include <hwlib.hpp>
+
+namespace r2d2::sound {
+    mp3_player_c::mp3_player_c(hwlib::pin_out &power_pin,
+                               hwlib::pin_oc &volume_up_next_pin,
```

Comment:
Why are these pins pin_oc? You are not reading from them. So they should be a pin_out

Reply:
This one has to do with the board. it didn't work with the pin_out for the small. That's the reason why the power pin is used as regular out. but the others didn't

Reply:
What do you mean with "didn't work for the small"?